pkg/reconciler/broker/broker_test.go

```diff
@@ -376,13 +459,23 @@ func TestAllCases(t *testing.T) {
 			drStore = NewDataresidencyTestStore(t, cm.(*corev1.ConfigMap))
 		}
 
+		// if maxPSClientCreateTime is in testData, no pubsub client is passed to reconciler, the reconciler
+		// will create one in demand
+		testPSClient := psclient
+		if maxTime, ok := testData["maxPSClientCreateTime"]; ok {
+			// Overwrite the createPubsubClientFn to one that failed when calling more than once, this does
+			// not really work since we never test anything that reconcile twice or both reconcile and delete
```

Comment:
If this doesn't work, should we keep it? Is it expected to work later, or is it fundamentally guaranteed to never work?

Reply:
Though the twice check does not work, it does still test that the pubsub client is created on demand, so it is still useful anyway. If later we can find the case to make sure it is never called twice it will be perfect.

Reply:
Makes sense. Can you add this explanation to the comment, so it's clear the code has a purpose?

Reply:
Make sense. Changed as suggested by moving the explanation to the test case.